plugin/example/source/dextool/plugin/cpptestdouble/runner.d

```diff
@@ -0,0 +1,150 @@
+/**
+Copyright: Copyright (c) 2017, Joakim Brännström. All rights reserved.
+License: MPL-2
+Author: Joakim Brännström (joakim.brannstrom@gmx.com)
+
+This Source Code Form is subject to the terms of the Mozilla Public License,
+v.2.0. If a copy of the MPL was not distributed with this file, You can obtain
+one at http://mozilla.org/MPL/2.0/.
+
+This file contains an example plugin.
+*/
+
+// for the generic main function to find the plugin the module must have this
+// name.
+module dextool.plugin.runner;
+
+import std.stdio;
+import logger = std.experimental.logger;
+
+import dextool.type : ExitStatusType;
+
+// called by the generic main function in standard.d
+ExitStatusType runPlugin(string[] args) {
+    RawConfiguration pargs;
+    pargs.parse(args);
+
+    // the dextool plugin architecture requires that two lines are printed upon
+    // request by the main function.
+    // a name of the plugin.
+    // a oneliner description.
+    if (pargs.shortPluginHelp) {
+        writeln("example");
+        writeln("plugin description");
+        return ExitStatusType.Ok;
+    } else if (pargs.help) {
+        pargs.printHelp;
+        return ExitStatusType.Ok;
+    } else if (pargs.file.length == 0) {
+        writeln("Missing file --in");
+        return ExitStatusType.Errors;
+    }
+
+    writeln("Run the plugin -d to see the AST");
+
+    import dextool.utility : prependDefaultFlags;
+
+    const auto cflags = prependDefaultFlags(pargs.cflags, "-xc++");
+
+    import std.typecons : Yes;
+    import cpptooling.analyzer.clang.context : ClangContext;
+
+    auto ctx = ClangContext(Yes.useInternalHeaders, Yes.prependParamSyntaxOnly);
+    auto visitor = new TUVisitor;
+
+    import dextool.utility : analyzeFile;
+
+    return analyzeFile(pargs.file, cflags, visitor, ctx);
+}
+
+// It is a good idea to separate the configuration parsing from the main function.
```

Comment:
If we knew what the plugin did, I think it would be easier to motivate the need for configuration.